communote/webapp/src/main/webapp/javascript/utils/AjaxFileUpload.js

```diff
@@ -58,12 +63,31 @@ var AjaxFileUpload = new Class({
         this.uploadIdBase = String.uniqueID();
         this.runningUploads = {};
     },
+    collectDataOfForm: function() {

```

Comment:
Kann auch:

```
collectDataOfForm: function() {
        var data = {};
        var inputElems = this.uploadForm.querySelectorAll('input');
        var responseParam = this.options.htmlResponseParameter;
        for (var i = 0; i < inputElems.length; i++) {
            var elem = inputElems[i];
            // skip file input
            if (elem.type !== 'file' && elem.name !== responseParam) {
                data[elem.name] = elem.value;
            }
     }
        return data;
  },
```

elem wird nicht außerhalb des loops verwendet, da ist es besser (für die IDE) den Scope nur innerhalb zu definieren.

Reply:
In JavaScript gibt es für ```var``` basierte Variablen Deklarationen keinen Block-Scope. Die Variablen am Anfang der Funktion zu deklarieren ist eine daraus abgeleitet Code Style Empfehlung (von Crockford). IDEs sollten damit eigentlich zurecht kommen.